cmd/snap/cmd_run.go

```diff
@@ -67,7 +67,7 @@ func (x *cmdRun) Execute(args []string) error {
 	if x.Hook != "" && x.Command != "" {
 		return fmt.Errorf("cannot use --hook and --command together")
 	}
-	if x.Revision != "" && x.Hook == "" {
+	if x.Revision != "unset" && x.Revision != "" && x.Hook == "" {
```

Comment:
Why do we need "unset" here?

For context, one of the reasons why I put "unset" there originally was because we were passing around unset revisions as if they were valid in awkward places, and it ended up as far as reaching the file system in some cases. So with that string we have a visual clue that something is not right. Typing in --revision=unset is one of those cases.. where's that string coming from?